arch/riscv/core/pmp/core_pmp.c

```diff
@@ -306,6 +314,10 @@ void z_riscv_pmp_add_dynamic(struct k_thread *thread,
 	}
 
 	__ASSERT((index < CONFIG_PMP_SLOT), "no free PMP entry\n");
+	CHECKIF(!(index < CONFIG_PMP_SLOT)) {
```

Comment:
I saw, most of the `CHECKIF` replaces the origin `__ASSERT`, but this one does not. I am curious why keeping `__ASSERT ` here.